aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -37,7 +37,7 @@ class DataRecord:
 
     def __init__(
         self,
-        idempotency_key: str,
+        idempotency_key: str = "",
```

Comment:
Are we sure we want this? It's probably a breaking change to make this optional.